sycl/source/detail/os_util.cpp

```diff
@@ -93,19 +112,27 @@ size_t OSUtil::getOSMemSize() {
   MemInfo.dwLength = sizeof(MemInfo);
   GlobalMemoryStatusEx(&MemInfo);
   return static_cast<size_t>(MemInfo.ullTotalPhys);
-#endif
+#elif defined(SYCL_RT_OS_DARWIN)
+  int64_t Size;
+  sysctlbyname("hw.memsize", &Size, nullptr, nullptr, 0);
+  return static_cast<size_t>(Size);
+#endif // SYCL_RT_OS
 }
 
 void *OSUtil::alignedAlloc(size_t Alignment, size_t NumBytes) {
 #if defined(SYCL_RT_OS_LINUX)
   return aligned_alloc(Alignment, NumBytes);
+#elif defined(SYCL_RT_OS_POSIX_SUPPORT)
+  void *Addr;
+  int ReturnCode = posix_memalign(&Addr, Alignment, NumBytes);
+  return (ReturnCode == 0) ? Addr : nullptr;
 #elif defined(SYCL_RT_OS_WINDOWS)
   return _aligned_malloc(NumBytes, Alignment);
 #endif
 }
 
 void OSUtil::alignedFree(void *Ptr) {
-#if defined(SYCL_RT_OS_LINUX)
+#if defined(SYCL_RT_OS_LINUX) || defined(SYCL_RT_OS_POSIX_SUPPORT)
```

Comment:
Isn't checking for SYCL_RT_OS_POSIX_SUPPORT is enough?

Reply:
It is important for the preprocessing in free to match that of malloc.  That way, if someone ever refactors the macros, they will not break the code.